src/lib/types.ts

```diff
@@ -0,0 +1,41 @@
+import type {
+	ReferenceElement,
+	FloatingElement,
+	Placement,
+	Strategy,
+	Middleware,
+	MiddlewareData
+} from '@floating-ui/dom';
+
+export interface UseFloatingOptions {
+	placement?: Placement;
```

Comment:
Not at my workstation to test - but let's consider JSDoc comments to reveal the type options for each interface key. Fingers cross Floating UI's imported types handle this for us though!

Reply:
@Hugos68 so here's what I was referring to above. Here's what the consuming user sees:

![Screenshot 2024-04-25 at 2 09 47 PM](https://github.com/skeletonlabs/floating-ui-svelte/assets/1509726/2d642c69-d22e-4ef9-9869-b6ead6717a3a)

Versus what values are actually allowed:

![Screenshot 2024-04-25 at 2 09 54 PM](https://github.com/skeletonlabs/floating-ui-svelte/assets/1509726/e38f60a5-a6c9-443c-b8bc-92a77037f205)

I'm calling out `Placement` here, but same goes for `Strategy`, etc.

I'm not sure the benefit of the `MaybeGetter` here, but it's impacting the DX.